provisioning/service/samples/getting started/EnrollmentGroupSample/EnrollmentGroupSample.cs

```diff
@@ -52,7 +52,10 @@ private async Task EnumerateRegistrationsInGroupAsync(QuerySpecification querySp
             Console.WriteLine($"\nCreating a query for registrations within group '{group.EnrollmentGroupId}'...");
             using Query registrationQuery = _provisioningServiceClient.CreateEnrollmentGroupRegistrationStateQuery(querySpecification, group.EnrollmentGroupId);
             Console.WriteLine($"\nQuerying the next registrations within group '{group.EnrollmentGroupId}'...");
-            QueryResult registrationQueryResult = await registrationQuery.NextAsync();
+            while (registrationQuery.HasNext()) {
+                QueryResult registrationQueryResult = await registrationQuery.NextAsync();
+                Console.WriteLine(registrationQueryResult.ToString());
+            }
```

Comment:
ToString() won't yield a meaningful value, I think. Give me a moment to propose a snippet.

Reply:
it dumps JSON:


`{  "Type": "DeviceRegistration",  "Items": [    {      "registrationId": "sampleDevice",      "createdDateTimeUtc": "2023-01-12T14:53:05.739285Z",      "lastUpdatedDateTimeUtc": "2023-01-12T14:53:06.0053269Z",      "assignedHub": "mcbdemohubs2.azure-devices.net",      "deviceId": "sampleDevice",      "status": "assigned",      "etag": "IjI3MDAyODc2LTAwMDAtMDEwMC0wMDAwLTYzYzAxZWQyMDAwMCI="    }  ],`

sorry, copy and paste is not picking up right, but you get the idea 🤣

Reply:
Wow, upon inspection this overall sample is just terrible with printing results. I'm going to make wider changes including improving doc comments on the SDK and will submit a separate PR because the scope of changes won't fit in as comments in a PR.

Thank you for bringing this to our attention.

Reply:
> it dumps JSON

Yeah, there is an overload on the objects for ToString, but still not useful as a sample for showing a user how to use the resulting object.

Reply:
ok then I guess the question is would you like to just close this as out of date now?  or leave it open as a reminder? either is fine :)

Reply:
Hang on until I get a PR out and then we can choose to close it. :)